undistort_example.py

```diff
@@ -0,0 +1,139 @@
+        o3d.pipelines.registration.TransformationEstimationPointToPoint(),  
+        o3d.pipelines.registration.ICPConvergenceCriteria(max_iteration=20000),
+    )
+
+    # returning initial pcd to it's original state
+    # this is done for correct sector cutting
+    pcd_initial.transform(np.linalg.inv(pose_start))
+
+
+    coefs = np.roll(coefs, -len(coefs)//2 ) # rolled due to lidar ray start
+    for i in range(1, N_segments):
+        # cutting a segment
+        segment = cut_segment(pcd_initial, sectors[i], start_angle,  coefs[i-1])
+
+        # getting composite transform matrix: initial -> pose --(gradual)--> target_pcd pose
+        transform_matrix = get_intermediate_transform(initial_pose, reg_p2p.transformation@pose_start, coefs[i-1])
+
+        # applying transform to the segment
+        undistort_result.append(segment.transform(transform_matrix))
+        
+        start_angle = sectors[i]
+
+    return undistort_result
+
+def read_pose(filename):
+    matrix = np.eye(4)
+    try:
+        f = open(filename, newline='')
+    except OSError:
+        print(f"Could not open/read file: {filename}. Default pose returned", filename)
+    with f:
+        lines = f.readlines()
+        matrix = np.asarray([[round(float(el), 4) for el in row.split(" ")[0:7]] for row in lines])
+        
+    return matrix
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--pcd_start",
+        type=str,
+        default='/home/polosatik/pointcloud_undistort/hilti_14/clouds/0.pcd',
+        help="pcd of initial pose",
+    )
+    parser.add_argument(
+        "--pcd_final",
+        type=str,
+        default='/home/polosatik/pointcloud_undistort/hilti_14/clouds/15.pcd',
+        help="pcd of final pose",
+    )
+
+    parser.add_argument(
+        "--pose_start",
+        type=str,
+        default='/home/polosatik/pointcloud_undistort/hilti_14/poses/0.txt',
+        help="pose of initial pose",
+    )
+    parser.add_argument(
+        "--pose_final",
+        type=str,
+        default='/home/polosatik/pointcloud_undistort/hilti_14/poses/15.txt',
+        help="pose of final pose",
+    )
+
+    parser.add_argument(
+        "--N_segments",
+        type=int,
+        default=8,
+        help="number of segments to split pointcloud into, must be an even number",
+    )
+
+
+    args = parser.parse_args()
+
+    pcd_start = o3d.io.read_point_cloud(str(Path(args.pcd_start)))
+    pcd_start = pcd_start.paint_uniform_color([1.0, 0.0, 0.0])
+
+    pcd_fin = o3d.io.read_point_cloud(str(Path(args.pcd_final)))
+    pcd_fin = pcd_fin.paint_uniform_color([0.0, 1.0, 0.0])
+
+    pcd_undistort = o3d.io.read_point_cloud(str(Path(args.pcd_start)))
+
+    pose_start = read_pose(Path(args.pose_start))
+    pose_fin = read_pose(Path(args.pose_final))
+
+    undistort_pcd = undistort_cloud(o3d.geometry.PointCloud(pcd_start), o3d.geometry.PointCloud(pcd_fin), pose_start, pose_fin, args.N_segments)
```

Comment:
It's completely useless casts `o3d.geometry.PointCloud(pcd_start)` like this

pcd_start and pcd_fin already point cloud types. Also bad naming for variables

Reply:
https://www.open3d.org/docs/release/python_api/open3d.geometry.PointCloud.html this is not a cast, but a copy constructor, due to the fact that objects are mutable, but moved copying in the function.